Readme.md

```diff
@@ -29,37 +12,44 @@ Firefox, Safari and Edge not supported as they are not yet properly implementing
 -   Edge: <https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/11963735/>
 -   Safari: <https://bugs.webkit.org/show_bug.cgi?id=171934>
 
-## Quick start
-
--   Clone repository, this will be the $GOPATH
-
--   In your editor settings, set $GOPATH to repo if needed
-
--   Install dependencies:
+## Development
 
-        make deps
+### Prerequisites
 
--   Build and run
+[Nix](https://nixos.org/nix) is required for installing dependencies and providing a suitable development environment.
 
-        make
-        ./bin/dividat-driver
+### Quick start
 
-    Or use `go run`:
+- Create a suitable environment: `nix-shell`
+- Build the driver: `make`
+- Run the driver: `./bin/dividat-driver`
 
-        go run src/cmd/dividat-driver/main.go start
-
-## Tests
+### Tests
 
 Run the test suite with: `make test`.
 
-## Releasing
+### Releasing
 
 Currently releases can only be made for Linux (from Linux). Builds on Linux are statically linked with [musl](https://www.musl-libc.org/).
 
 To create a release run: `make release`. You will need to be able to provide appropriate signing keys.
 
 To deploy a new release run: `make deploy`. This can only be done if you are on `master` or `develop` branch, have correctly tagged the revision and have AWS credentials set in your environment.
 
+### Go packages
+
+Go dependencies are provided by the [Go machinery](https://nixos.org/nixpkgs/manual/#sec-language-go) in Nix.
+
+For local development you may use `dep` to install go dependencies: `cd src/dividat-driver && dep ensure`.
+
+New Go dependencies can be added with `dep` (e.g. `dep ensure -add github.com/something/supercool`). The specification of dependencies needs to be recreated for Nix. This can be done with the command `make nix/deps.nix`. Check in the updated `Gopkg.toml`, `Gopkg.lock` and `nix/deps.nix` files.
```

Comment:
Let's just make `crossbuild` (or rather the individual LINUX_BIN etc. targets) depend on `nix/deps.nix`?

Reply:
I prefer having the default environment just being available. Usage of dep is not hindered in any way by providing already existing go packages via `nix/deps.nix`. Just makes it faster and easier to get started (on my machine `nix` is considerably faster than `dep ensure`).

Reply:
I'm okay with checking it in if there is a way to ensure that it stays up to date. I don't like these Readme side conditions on the build process that are easy to overlook or forget.

Reply:
[Git hooks](https://git-scm.com/book/gr/v2/Customizing-Git-Git-Hooks) could be useful, but there doesn't seem to be a good way for sharing them.

Reply:
And checked in or not: if we remove phoniness from the underlying targets, expressing the dependency on `nix/deps.nix` does not hurt execution time, but prevents confusion.

Reply:
Very good point. And yup, it is very easily automatizable by adding `nix/deps.nix` as a dependency of the `build` target:

https://github.com/dividat/driver/pull/85/commits/537ed0d40a35aa3763e2d9dd43fd49a65f78d76d#diff-b67911656ef5d18c4ae36cb6741b7965L44

 Now on every build `nix/deps.nix` is regenerated if `src/dividat-driver/Gopkgs.toml` has been changed. Readme updated.